source/auth/tests/mongodb-oidc.rst

```diff
@@ -130,6 +130,66 @@ credentials.
 #. Create a new client using the AWS device workflow.
 #. Ensure that a ``find`` operation does not add credentials to the cache.
 
+Speculative Authentication
+==========================
+
+#. Clear the cache.
+#. Create a client with a request callback that returns a valid token
+  that will not expire soon.
+#. Set a fail point for ``saslStart`` commands of the form:
+
+.. code:: javascript
+
+    {
+      "configureFailPoint": "failCommand",
+      "mode": {
+        "times": 2
+      },
+      "data": {
+        "failCommands": [
+          "saslStart"
+        ],
+        "errorCode": 18
+      }
+    }
+
+Note: the driver MUST either use a unique ``appName`` or explicitly
+remove the ``failCommand`` after the test to prevent leakage.
+
+#. Perform a ``find`` operation.
+#. Close the client.
+#. Create a new client with the same properties.
```

Comment:
Do we need to do this repeatedly? It seems like if the first find succeeds we know the speculative auth was in the initial handshake correct?